### ISSUE TITLE
[BUG] App crashes when new Intent is launched and no app can handle it

### ISSUE THREAD

**Issue by fesave** [owncloud/android#3581]
### Actual behaviour

When you try to send feedback from the navigation drawer and your device does not have any application

- [x] to send an email
- [x] to authenticate with the browser

the application crashes.

### Expected behaviour

The application should not crash or should warn the user that there is no application available to perform this action.

 
### Steps to reproduce
1. Check that your device does not have any e-mail application.
2. Try to send feedback from the navigation drawer.

**Comment by fesave** [comment]
Fixed in #3512

**Comment by jesmrec** [comment]
### (1)

In a device with no browser available (or in a work profile without browser):

1. Install the app from scratch
2. Enter an OAuth2 server url (f.ex. demo.owncloud.com)
3. Click on `Log in` button

Current: app crashes because no browser is there. Stacktrace:

```
2022-03-30 16:52:24.474 4531-4531/com.owncloud.android.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.owncloud.android.debug, PID: 4531
    android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=https://demo.owncloud.com/... (has extras) }
        at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2067)
        at android.app.Instrumentation.execStartActivity(Instrumentation.java:1727)
        at android.app.Activity.startActivityForResult(Activity.java:5320)
        at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:597)
        at android.app.Activity.startActivityForResult(Activity.java:5278)
        at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:583)
        at android.app.Activity.startActivity(Activity.java:5664)
        at androidx.core.content.ContextCompat$Api16Impl.startActivity(ContextCompat.java:830)
        at androidx.core.content.ContextCompat.startActivity(ContextCompat.java:279)
        at androidx.browser.customtabs.CustomTabsIntent.launchUrl(CustomTabsIntent.java:376)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity.performGetAuthorizationCodeRequest(LoginActivity.kt:461)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity.performGetAuthorizationCodeRequest$default(LoginActivity.kt:442)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity.startNormalOauthorization(LoginActivity.kt:439)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity.startOIDCOauthorization$lambda-30(LoginActivity.kt:398)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity.$r8$lambda$OGg6bj0JO23Wp3EKfuKLc17iV8U(Unknown Source:0)
        at com.owncloud.android.presentation.ui.authentication.LoginActivity$$ExternalSyntheticLambda10.onChanged(Unknown Source:4)
        at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
        at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
        at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
        at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
        at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

Expected: notify user that no browser is available

Pixel2 Android11
`d7663b180`

**Comment by jesmrec** [comment]
### (2)

Behaviour should be unified through all features for the same steps.

Screenshot 1: Opening a log file in a device without a proper app -> Snackbar
Screenshot 2: Send Feedback in a device without a proper app -> Toast

Screenshot 1 | Screenshot2
-|-
|![Screenshot_20220330_165631](https://user-images.githubusercontent.com/14894746/160865615-31a3ac95-e3e7-47c8-b8aa-97828f5820f3.png)|![Screenshot_20220330_165647](https://user-images.githubusercontent.com/14894746/160865648-6d900e17-df09-41b7-b16f-65ece0552322.png)

Error is the same with different text and presentation. 
If i'd go for one, i will take the snackbar.

Pixel2 Android11
`d7663b180`

**Comment by fesave** [comment]
Fixed both of them @jesmrec 😉

**Comment by jesmrec** [comment]
(1) and (2) fixed

**Comment by jesmrec** [comment]
This is OK and done, will be merged inside #3512
